ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -156,13 +156,22 @@ public LogicalSchema withPseudoAndKeyColsInValue(final boolean windowed) {
     return withPseudoAndKeyColsInValue(windowed, CURRENT_PSEUDOCOLUMN_VERSION_NUMBER);
   }
 
+  /**
+   * Remove pseudo and key columns, according to the pseudocolumn version from the value schema.
+   * @param pseudoColumnVersion the version of pseudocolumns to evaluate against
+   * @return the new schema with the columns removed
+   */
+  public LogicalSchema withoutPseudoAndKeyColsInValue(final int pseudoColumnVersion) {
```

Comment:
Does this new method need to be public?

Reply:
I'm using it in `LogicalSchemaTest` at least, and I assumed it'd likely need to be used down the line in some of the places it's 0-arg overload is used (as I thought we discussed in our last sync). What's your thought process on why it wouldn't be public?

Reply:
`LogicalSchemaTest` is in the same package so package-private is good enough for that use case. In general, it's easier to make things public later when it's determined that they need to be public than the other way around.

Reply:
That makes sense, thanks!